test/faker/date_test.exs

```diff
@@ -35,9 +35,16 @@ defmodule DateTest do
   end
 
   defp age(%Date{year: year, month: month, day: day}) do
-    %Date{ year: current_year, month: current_month, day: current_day } = now()
-    already_aged_this_year = current_month > month || current_month == month && day >= current_day
-    current_year - year + (if already_aged_this_year, do: 1, else: 0)
+    %Date{year: current_year, month: current_month, day: current_day} = now()
+    min_age = current_year - year - 1
+
+    had_birthday_this_year =
+      current_month > month || (current_month == month && current_day >= day)
+
+    case had_birthday_this_year do
+      true -> min_age + 1
+      false -> min_age
+    end
```

Comment:
I'm kind of scared to see both the test & implementation changed. Was there something wrong with the test or didn't you like the messy inline stuff?

Reply:
Ah yes, i should have commented on that.
The test assumed that `current_year - year_birth` is the minimum age of a person, and added 1 to that if the person already had a birthday this year. This would mean that someone born in 2016 would be 1 or 2 years old in 2017, while it's really 0 or 1.
Also, in `day >= current_day` was the wrong day around.

Example:
```
iex(29)> age(~D[2016-10-30])
2
iex(30)> age(~D[2016-10-20])
1
iex(31)> age(~D[2017-10-20])
0
iex(32)> age(~D[2017-10-30])
1
```